Added gpas-worker chef-cookbook

bilalbaqar opened the pull request.

bilalbaqar committed:
Added ec2-slurm-worker cookbook

bilalbaqar committed:
enamed cookbook to gpas-worker

bilalbaqar committed:
Added detect-secrets with pre-commit

bilalbaqar committed:
Cleaned up old cookbook

bilalbaqar committed:
Added upstream cookbooks

singergr-ctds commented on packer/amzn2/packer.json:
```diff
@@ -0,0 +1,70 @@
+    "instance_type": "t3.micro",
+    "ssh_username": "ec2-user",
+    "ami_name": "packer-gpas-amzn2_{{isotime | clean_resource_name}}",
+    "subnet_id": "subnet-0dc5a5bc9b0033a6b",
```
It might be nice to replace some of these with [variables](https://www.packer.io/docs/templates/user-variables.html). That can happen in a follow-up PR, though—I think anyone familiar with Packer can work this out.

singergr-ctds commented on packer/cookbooks/gpas-worker/files/default/mountEphemeral.sh:
```diff
@@ -0,0 +1,33 @@
+  sudo mount LABEL=EPHEMERAL_RAID /mnt
+fi
+
+mkdir -p /mnt/SCRATCH/slurm_automation_logs
```
Do we have any workflow dependencies on the existence of the Slurm-named directories? Just curious.

bilalbaqar replied:
Nope. I missed this. Thanks for pointing it out.

singergr-ctds commented on packer/cookbooks/gpas-worker/files/default/mountEphemeral.sh:
```diff
@@ -0,0 +1,33 @@
+
+mkdir -p /mnt/SCRATCH/slurm_automation_logs
+
+chown -R slurm:slurm /mnt/SCRATCH
```
Probably don't need this if we're not installing Slurm. I'm not sure which user should own the directory.

bilalbaqar replied:
Yup. good catch.

singergr-ctds commented on packer/cookbooks/gpas-worker/recipes/python.rb:
```diff
@@ -0,0 +1,86 @@
+  not_if "grep -q virtualenvwrapper.sh #{home_dir}/.bashrc"
+end
+
+%w(SQLAlchemy psycopg2 awscli==1.10.14 pysam==0.15.2 cwltool==1.0.20180306163216).each do |package|
```
cwltool version numbering. 😅

singergr-ctds reviewed (approved):
I noted a couple things, but I think it's fine to merge and update more later if needed.

bilalbaqar committed:
Updated packer.json to support variables

bilalbaqar committed:
Updated README.md for packer

bilalbaqar committed:
Merge branch 'feat/chef-cookbook' into feat/chef-upstream-cookbooks

bilalbaqar committed:
Added seven_zip and windows cookbooks

bilalbaqar committed:
Added upstream cookbooks

Upstream cookbooks added:
- apt
- ark
- os-hardening
- seven_zip
- windows
- yum-epel
